fuel-vm/src/interpreter/memory.rs

```diff
@@ -73,96 +377,55 @@ impl ToAddr for i32 {
+    /// Create a new memory range. Cannot panic, but the range may be invalid.
+    pub const fn new(start: usize, len: usize) -> Self {
+        Self(start..start.saturating_add(len))
     }
-}
-
-impl ops::Deref for MemoryRange {
-    type Target = ops::Range<usize>;
 
-    fn deref(&self) -> &Self::Target {
-        &self.0
+    /// Start of the range.
+    pub fn start(&self) -> usize {
+        self.0.start
     }
-}
-
-impl MemoryRange {
-    /// Create a new memory range represented as `[address, address + size[`.
-    pub fn new<A: ToAddr, B: ToAddr>(address: A, size: B) -> Result<Self, PanicReason> {
-        let start = address.to_addr()?;
-        let size = size.to_addr()?;
-        let end = start.checked_add(size).ok_or(PanicReason::MemoryOverflow)?;
-
-        if end > MEM_SIZE {
-            return Err(PanicReason::MemoryOverflow)
-        }
 
-        Ok(Self(start..end))
+    /// End of the range. One past the last byte.
+    pub fn end(&self) -> usize {
+        self.0.end
     }
 
-    /// Create a new const sized memory range.
-    pub fn new_const<A: ToAddr, const SIZE: usize>(
-        address: A,
-    ) -> Result<Self, PanicReason> {
-        Self::new(address, SIZE)
+    /// Is the range empty?
+    pub fn is_empty(&self) -> bool {
+        self.len() == 0
     }
 
-    /// Uses a overflow-capturing operator to combine `base` and `offset`
-    /// into the start of a memory range, returning an error on overflow,
-    /// and then checks that the resulting range is within the VM memory.
-    pub fn new_overflowing_op<A: ToAddr, T: ToAddr, F>(
-        overflowing_op: F,
-        base: T,
-        offset: T,
-        size: A,
-    ) -> Result<Self, PanicReason>
-    where
-        F: FnOnce(T, T) -> (T, bool),
-    {
-        let (addr, overflow) = overflowing_op(base, offset);
-        if overflow {
-            return Err(PanicReason::MemoryOverflow)
-        }
-        Self::new(addr, size)
+    /// Length of the range.
+    pub fn len(&self) -> usize {
+        self.0.len()
     }
 
-    /// Truncate a memory range to a new size if it's smaller then current one.
-    pub fn truncated(&self, limit: usize) -> Self {
-        Self::new(self.start, self.len().min(limit))
-            .expect("Cannot overflow on truncation")
+    /// Returns the range as a `usize` range.
+    pub fn usizes(&self) -> Range<usize> {
+        self.0.clone()
     }
 
-    /// Return `true` if the length is `0`.
-    pub fn is_empty(&self) -> bool {
-        self.len() == 0
+    /// Returns the range as a `Word` range.
+    pub fn words(&self) -> Range<Word> {
+        self.0.start as Word..self.0.end as Word
     }
 
-    /// Convert to a raw `usize` range.
-    pub fn usizes(&self) -> ops::Range<usize> {
-        self.start..self.end
-    }
+    /// Moves range to be relative to the given address.
+    /// Panics if the address is before the offset.
+    pub fn relative_to(self, index: usize) -> Self {
+        assert!(self.start() >= index);
```

Comment:
We use this function to places where we verify that `start >= index`, so it shouldn't fail. But since this function is public, I would prefer to remove this function and use:

```rust
let start = self.start() - index;
let end = self.end() - index;
```

Directly in the place of this function usage=)

Reply:
I prefer having the function, but we can make it private.

Reply:
Ah nevermind, just removed/inlined the function in ad312a8b